pkg/pool-stable/test/ComposableStablePool.test.ts

```diff
@@ -605,6 +607,124 @@ describe('ComposableStablePool', () => {
       });
     });
 
+    describe('beforeJoinExit', () => {
+      let registeredBalances: BigNumber[];
+
+      sharedBeforeEach('deploy and initialize pool', async () => {
+        await deployPool({ admin });
+        await pool.init({ initialBalances, recipient: lp });
+        registeredBalances = await pool.getBalances();
+      });
+
+      function itPaysProtocolFeesAndReturnsNecessaryData() {
+        context('preJoinExitSupply', () => {
+          context('when protocol fees are due', () => {
+            // This is tested more comprehensively in ComposableStablePoolProtocolFees.
+            // Here we just need to check that _beforeJoinExit calls into _payProtocolFeesBeforeJoinExit.
+            let registeredBalancesWithFees: BigNumber[];
+            let expectedBptAmount: BigNumber;
+
+            const FEE_RELATIVE_ERROR = 1e-3;
+            sharedBeforeEach(async () => {
+              const MIN_SWAP_BALANCE_DELTA = 1.5e3;
+              const MAX_SWAP_BALANCE_DELTA = 20e3;
+              const deltas = range(numberOfTokens + 1).map((_, i) =>
+                i !== bptIndex ? fp(random(MIN_SWAP_BALANCE_DELTA, MAX_SWAP_BALANCE_DELTA)) : 0
+              );
+
+              registeredBalancesWithFees = arrayAdd(registeredBalances, deltas);
+
+              // We assume all tokens have similar value, and simply add all of the amounts together to represent how
+              // much value is being added to the Pool. This is equivalent to assuming the invariant is the sum of the
+              // tokens (which is a close approximation while the Pool is balanced).
+
+              const deltaSum = bnSum(deltas);
+              const currSum = bnSum(registeredBalancesWithFees);
+              const poolPercentageDueToDeltas = deltaSum.mul(FP_SCALING_FACTOR).div(currSum);
+
+              const protocolSwapFeesPercentage = fp(0.5);
```

Comment:
Don't you need for this value to be set globally and the pool to update its cache to get it to be used?